prov/sm2/src/sm2_progress.c

```diff
@@ -351,8 +438,9 @@ void sm2_progress_recv(struct sm2_ep *ep)
 					0, atomic_entry->atomic_data.data,
 					xfer_entry->hdr.size);
 			}
-			if (xfer_entry->hdr.op_flags & FI_DELIVERY_COMPLETE &&
-			    !(xfer_entry->hdr.proto_flags & SM2_UNEXP)) {
+
+			if (xfer_entry->hdr.proto_flags &
```

Comment:
I would really like to hide the return handling logic in a static inline function for the following reasons:
1. it looks horrible with the 80 char line length limit
2. I think it makes the sm2 progress loop go from readable to very unclean.

Reply:
done